src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Prerendering.cs

```diff
@@ -81,6 +82,15 @@ public static void MarkAsAllowingEnhancedNavigation(HttpContext context)
         context.Response.Headers.Add("blazor-enhanced-nav", "allow");
     }
 
+    public static void ProcessEnhancedNavigationReplaceHistoryCookie(HttpContext context)
+    {
+        if (context.Request.Cookies.ContainsKey(_enhancedNavReplaceHistory))
+        {
+            context.Response.Headers.Add(_enhancedNavReplaceHistory, "true");
+            context.Response.Cookies.Delete(_enhancedNavReplaceHistory);
+        }
```

Comment:
Unfortunately that's not really true. `redirect: manual` does **not** do what everyone thinks it will do. In fact, unless you're writing a service worker, there's basically nothing useful that it does at all. See https://stackoverflow.com/questions/42716082/fetch-api-whats-the-use-of-redirect-manual for more info, or generally search for the hundreds of posts across the web where people really cannot believe that `redirect: manual` is this unhelpful, but it actually is :)